Assets/SteamVR_Unity_Toolkit/Scripts/VRTK_PlayerClimb.cs

```diff
@@ -0,0 +1,188 @@
+﻿namespace VRTK
+{
+    using UnityEngine;
+    using System.Collections;
+
+    public struct PlayerClimbEventArgs
+    {
+        public uint controllerIndex;
+        public GameObject target;
+    }
+
+    public delegate void PlayerClimbEventHandler(object sender, PlayerClimbEventArgs e);
+
+    public class VRTK_PlayerClimb : MonoBehaviour
+    {
+        public event PlayerClimbEventHandler PlayerClimbStarted;
+        public event PlayerClimbEventHandler PlayerClimbEnded;
+
+        public bool usePlayerScale = true;
+        public bool useGravity = true;
+
+        private Transform controllerTransform;
+        private Vector3 startControllerPosition;
+        private Vector3 startPosition;
+
+        private Vector3 lastGoodHeadsetPosition;
+        private bool headsetColliding = false;
+        private bool isClimbing = false;
+
+        VRTK_PlayerPhysics playerPhysics;
+        private bool lastGravitySetting;
+
+        public virtual void OnPlayerClimbStarted(PlayerClimbEventArgs e)
+        {
+            if (PlayerClimbStarted != null)
+                PlayerClimbStarted(this, e);
+        }
+
+        public virtual void OnPlayerClimbEnded(PlayerClimbEventArgs e)
+        {
+            if (PlayerClimbEnded != null)
+                PlayerClimbEnded(this, e);
+        }
+
+        protected PlayerClimbEventArgs SetPlayerClimbEvent(uint controllerIndex, GameObject target)
+        {
+            PlayerClimbEventArgs e;
+            e.controllerIndex = controllerIndex;
+            e.target = target;
+            return e;
+        }
+
+        private void Start()
+        {
+            var controllerManager = GameObject.FindObjectOfType<SteamVR_ControllerManager>();
+            InitControllerListeners(controllerManager.left);
+            InitControllerListeners(controllerManager.right);
+
+            // Listen for teleport events 
+            VRTK_BasicTeleport teleportComponent = GetComponent<VRTK_BasicTeleport>();
+            if (teleportComponent)
+            {
+                teleportComponent.Teleporting += new TeleportEventHandler(OnTeleport);
+            } 
+
+            // Required Components
+
+            // VRTK_PlayerPhysics
+            playerPhysics = GetComponent<VRTK_PlayerPhysics>();
+            if (!playerPhysics && useGravity)
+            {
+                playerPhysics = gameObject.AddComponent<VRTK_PlayerPhysics>();
+            }
+
+            // VRTK_HeadsetCollisionFade
+            Transform headCamera = GameObject.FindObjectOfType<SteamVR_GameView>().GetComponent<Transform>();
+            VRTK_HeadsetCollisionFade collisionFade = headCamera.GetComponent<VRTK_HeadsetCollisionFade>();
```

Comment:
I'd var this as well

`var collisionFade = headCamera.GetComponent<VRTK_HeadsetCollisionFade>();`